### ISSUE TITLE
js error from certain html

### ISSUE THREAD

**Issue by philfreo** [froala/wysiwyg-editor#396]
http://jsfiddle.net/65o11ap2/

**Comment by philfreo** [comment]
still breaks as soon as you enter HTML mode

![screenshot 2015-01-17 06 57 31](https://cloud.githubusercontent.com/assets/97612/5789206/1dfe2060-9e16-11e4-8598-53b0f6904c9f.png)

:(

**Comment by stefanneculai** [comment]
How have you got to initialize the editor with that specific HTML?

**Comment by philfreo** [comment]
We're using Froala as an html email compose tool, and when replying to an existing HTML message it may contain crap like this from Outlook-generated emails, etc and we blockquote the previous message. I know blockquoting an `<html>` tag, etc. isn't technically valid HTML, but generally it works just fine and I expected Froala's cleanup code to gracefully handle cases like this.

**Comment by stefanneculai** [comment]
If you want to remove the comments, you should use the option `allowComments: false`. By default it is set to `true` because there are a lot of users who want to allow comments.

**Comment by philfreo** [comment]
http://jsfiddle.net/axpq8eu2/

Click the button and notice how the styles on the rest of the page get messed up

**Comment by stefanneculai** [comment]
No, it does what it should. There is some style inside the HTML that is set to the editor that changes the content outside of the editor. 

You have 2 options: 
- do something similar to http://jsfiddle.net/2k0rpjz1/
- use the editor within an iframe so that the style inside it is not affected by the style from the rest of the page

**Comment by philfreo** [comment]
My basic assumption is:

A froala editor should be able to be passed any (potentially untrusted) input content and a) not choke on the content, b) be safe for the user / prevents XSS, c) doesn't affect anything outside of the editor

Is there an option to make Froala use an iframe, if that's what's necessary to satisfy (c)?

**Comment by stefanneculai** [comment]
a), b) It does that based on what options that are passed. If you allow style tag style by default, then adding style in HTML mode might affect what is displayed as it is expected. As you can see in the jsFiddle from http://jsfiddle.net/2k0rpjz1/ it doesn't add the style if you don't include it in the allowedTags list. It would make sense to remove the style be default. We'll add that in.

c) We are already working on an iframe option. However, it won't be available for sure until version 2.0.0.

**Comment by philfreo** [comment]
Very helpful, thanks. I'll remove `<style>` from `allowedTags` and turn off `allowComments` (even though I don't think this should be necessary). I'm glad you're changing the default though - probably the safest for everyone as a default.

**Comment by stefanneculai** [comment]
See f53d4dbc3eeb4c60e813267a016eac1562a2519a. We haven't thought that there might be users who would input other bad `<style>` to break the rest of the page and haven't considered it an XSS so far. Now it is removed by default and that can't be changed.

**Comment by philfreo** [comment]
Another similar case is still breaking it:

http://jsfiddle.net/91ekzse5/

**Comment by philfreo** [comment]
Any ideas?

**Comment by stefanneculai** [comment]
We have it fixed under dev. We have to run few more tests and will put the update here

**Comment by philfreo** [comment]
Should I upgrade now?

**Comment by stefanneculai** [comment]
Yes, it's fixed.

**Comment by philfreo** [comment]
Thanks!

**Comment by philfreo** [comment]
Hey @stefanneculai I noticed that https://froala.com/wysiwyg-editor/docs/options#allowedTags still shows `style` in the default list of allowedTags. Is this an error in the documentation or what? (You mentioned above removing it from the default list).

Also, I noticed that `base` can be used for some pretty nasty stuff (changing the location of AJAX requests on the page). Maybe you want to remove that from the default list as well?
